api/user-story/models/user-story.js

```diff
@@ -6,24 +6,34 @@
  */
 
 module.exports = {
-    lifecycles: {
-        async afterUpdate(result) {
-            try {
-                if(result.user_story_status.Status === 'Deployed') {
-                    let users = result.followers.map(follower => follower.id)
-                    if(!users.includes(result.author.id)) {
-                        users.push(result.author.id)
-                    }
-                    await strapi.services['user-story-notification'].create({
-                        message: `${result.Title} has been launched`,
-                        users: users,
-                        seen: false,
-                        date: new Date(),
-                        link: `story/${result.id}`
-                    })
-                }
-            }
-            catch(e) {}
+  lifecycles: {
+    async afterUpdate(result) {
+      try {
+        if(result.user_story_status.Status === 'Deployed') {
+          let users = result.followers.map(follower => follower.id)
+          if(!users.includes(result.author.id)) {
+            users.push(result.author.id)
+          }
+          await strapi.services['user-story-notification'].create({
+            message: `${result.Title} has been launched`,
+            users: users,
+            seen: false,
+            date: new Date(),
+            link: `story/${result.id}`
+          })
         }
+      }
+      catch(e) {}
+    },
+    async beforeCreate(data) {
+      if (!data.user_story_status) {
```

Comment:
What is the need of if statement here?

Reply:
Actually in the seed data for testing, I had assigned status ids directly. So without the if, all the stories were assigned 'Under consideration' status.